datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxHorizontalScalingService.java

```diff
@@ -0,0 +1,133 @@
+            return ThreadBasedUserCrnProvider.doAsInternalActor(
+                    regionAwareInternalCrnGeneratorFactory.iam().getInternalCrnForServiceAsString(),
+                    () -> stackV4Endpoint.putScaling(0L, sdxCluster.getClusterName(), scaleRequest, sdxCluster.getAccountId())
+            ).getPollableId();
+        } catch (javax.ws.rs.NotFoundException e) {
+            LOGGER.info("Cannot find stack on Cloudbreak side {}", sdxCluster.getClusterName());
+            throw new RuntimeException("Cannot find stack on Cloudbreak side.", e);
+        } catch (WebApplicationException e) {
+            String errorMessage = webApplicationExceptionMessageExtractor.getErrorMessage(e);
+            LOGGER.info("Cannot horizontal scale stack {} from cloudbreak: {}", sdxCluster.getStackId(), errorMessage, e);
+            throw new RuntimeException(errorMessage);
+        } catch (ProcessingException e) {
+            throw new RuntimeException(e.getMessage());
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void validateHorizontalScaleRequest(SdxCluster sdxCluster, DatalakeHorizontalScaleRequest request) {
+        if (!ENTERPRISE.equals(sdxCluster.getClusterShape())) {
+            throw new BadRequestException(String.format("scaling not supported on: %s.", sdxCluster.getClusterShape().name()));
+        }
+        String hostGroup = request.getGroup();
+        if (!DatalakeInstanceGroupScalingDetails.valueOf(hostGroup.toUpperCase(Locale.ROOT)).isScalable()) {
+            throw new BadRequestException(String.format("%s host group cannot scale", hostGroup));
+        }
+        DetailedEnvironmentResponse environmentResponse = environmentClientService.getByName(sdxCluster.getEnvName());
+        if (!EnvironmentStatus.upscalable().contains(environmentResponse.getEnvironmentStatus())) {
+            throw new BadRequestException(String.format("The cluster cannot be scaled in the %s environment status.",
+                    environmentResponse.getEnvironmentStatus().name()));
+        }
+        StackV4Response stackResponse = stackV4Endpoint.getByCrn(1L, sdxCluster.getStackCrn(), Set.of());
+        if (!isNodeRequestValid(stackResponse, request)) {
+            throw new BadRequestException("The requested nodeCount does not reach the minimum condition of horizontal scaling.");
+        }
+    }
+
+    private boolean isNodeRequestValid(StackV4Response stack, DatalakeHorizontalScaleRequest request) {
+        if (request.getDesiredCount() < 0) {
+            LOGGER.warn("Negative nodeCount was requested.");
+            return false;
+        }
+        String target = request.getGroup();
+        boolean hostGroupNotExists = stack.getInstanceGroups().stream()
+                .map(InstanceGroupV4Response::getName)
+                .noneMatch(target::equalsIgnoreCase);
+        if (hostGroupNotExists) {
+            LOGGER.warn("The requested hostgroup name not found!");
```

Comment:
Would be nice to see these detailed cases in the bad request message other than just the general.
"The requested nodeCount does not reach the minimum condition of horizontal scaling"